src/Hosting/Hosting/test/HostingApplicationTests.cs

```diff
@@ -56,6 +56,51 @@ public void CreateContextWithEnabledLoggerCreatesActivityAndSetsActivityIdInScop
             Assert.Equal(Activity.Current.Id, pairs["ActivityId"].ToString());
         }
 
+        [Fact]
+        public void ActivityStopDoesNotFireIfNoListenerAttachedForStart()
+        {
+            // Arrange
+            var diagnosticSource = new DiagnosticListener("DummySource");
+            var logger = new LoggerWithScopes(isEnabled: true);
+            var hostingApplication = CreateApplication(out var features, diagnosticSource: diagnosticSource, logger: logger);
+            var startFired = false;
+            var stopFired = false;
+
+            diagnosticSource.Subscribe(new CallbackDiagnosticListener(pair =>
+            {
+                // This should not fire
+                if (pair.Key == "Microsoft.AspNetCore.Hosting.HttpRequestIn.Start")
```

Comment:
Not a fan of non-shared strings. Since we return false from the predicate below, no events will fire so we could just set a single bool in the subscribe func and check that?

Reply:
I initially had that but I wanted it to be specific.